### PR TITLE
Issue #4725: fix all violations from the tenth part of idea violations

### PR DESCRIPTION
**Pull request by Nimfadora**
Issue #4725

fix includes:
```
GjsLint
JSHint
JSLint
TsLint
Eslint
Jscs
InnerHTMLJS
Duplicates
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchAndThrowUtils.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    private IllegalCatchAndThrowUtils() {
+    }
+
+    /**
+     * Create set of class names and short class names.
+     *
+     * @param classNames array of class names.
+     * @return set of class names and short class names.
+     */
+    public static Set<String> parseClassNames(final String... classNames) {
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchAndThrowUtils.java]: please move this method to CheckUtils.java

Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchAndThrowUtils.java]: also remove `final` from parameter.

Comment (Nimfadora) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalCatchAndThrowUtils.java]: fixed

File: pom.xml [hunk 2]
```diff
@@ -2259,6 +2259,7 @@
               </targetClasses>
               <targetTests>
                 <param>com.puppycrawl.tools.checkstyle.utils.*</param>
+                <param>com.puppycrawl.tools.checkstyle.checks.coding.*</param>
```
Comment (romani) [inline pom.xml]: what for this update ?

File: src/site/resources/js/anchors.js [hunk 3]
```diff
@@ -1,23 +1,27 @@
-'use strict';
+/*jslint browser: true*/
+/*global window */
+(function () {
+    "use strict";
+    window.addEventListener("load", function () {
```
Comment (romani) [inline src/site/resources/js/anchors.js]: please confirm that you tested this update.

File: pom.xml (hunk 2 above)
Comment (Nimfadora) [inline pom.xml]: I moved method from coding check, to utils class. Utils class is in different pitest profile, so it led to downgrade of pitest coverage. I had to add this line and one extra test, to bring it back to 100%

File: src/site/resources/js/anchors.js (hunk 3 above)
Comment (Nimfadora) [inline src/site/resources/js/anchors.js]: Confirm

File: pom.xml (hunk 2 above)
Comment (Nimfadora) [inline pom.xml]: Time of pitest run for utils profile is still around 2-3 mins

**Pull request merged**